content/docs/user-guide/dvc-files.md

```diff
@@ -240,14 +248,245 @@ stages:
 ```
 
 Stage commands are listed again in `dvc.lock`, in order to know when their
-definitions change in the `dvc.yaml` file.
+definitions change in `dvc.yaml`.
 
 Regular <abbr>dependencies</abbr> and all kinds of <abbr>outputs</abbr>
 (including [metrics](/doc/command-reference/metrics) and
 [plots](/doc/command-reference/plots) files) are also listed (per stage) in
-`dvc.lock`, but with an additional field to store the hash value of each file or
-directory tracked by DVC. Specifically: `md5`, `etag`, or `checksum` (same as in
-`deps` and `outs` entries of `.dvc` files).
+`dvc.lock`, but with an additional field with a hash of their last known
+contents. Specifically: `md5`, `etag`, or `checksum` are used (same as in `deps`
+and `outs` entries of `.dvc` files).
+
+Full <abbr>parameter dependencies</abbr> (key and value) are listed too (under
+`params`), grouped by parameters file. And in the case of
+templated/parameterized `dvc.yaml` files (see next section), their actual values
+are substituted into the `dvc.lock` YAML structure.
+
+### Parameterize `dvc.yaml`
+
+`dvc.yaml` supports a templating format to reuse values from different sources
+in the YAML structure itself. The sources can be external
+[parameters files](/doc/command-reference/params), or internal `vars`, as
+detailed below.
```

Comment:
`reuse values` - why `reuse`? I'm not sure I understand this ...

`The sources can be external
[parameters files](/doc/command-reference/params), or internal `vars`, as
detailed below.` 

this explanation is too heavy, implicit, abrupt ... start from simple ideas and example, provide details later

Reply:
> why reuse?

You don't have to type the same value over and over.

> explanation is too heavy, implicit, abrupt

2 sentences is too much for an intro? This is how it looks:

![image](https://user-images.githubusercontent.com/1477535/103981524-90468280-5147-11eb-9be0-6601859dea1d.png)

I would just be removing the last sentence and put that info somewhere later. It seems mostly a stylistic decision to me.

Reply:
re `reuse` - I changed it to "insert" now.